frontend/src/components/NavDropdown.tsx

```diff
@@ -31,37 +22,61 @@ export default function NavDropdown({
   menuItems,
   setActiveNavDropdownIndex,
 }: NavDropdownProps): JSX.Element {
-  let isOpen = activeNavDropdownIndex === index;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function eventHandler(e: any) {
+    const dropdowns = document.getElementsByName("navDropDownButton");
```

Comment:
Doing a dom query here feels a little icky. I wonder if we could get around having to do a query like this by using refs, or even running this function at a higher level that knows about all of the dropdowns, and passing it through.

Reply:
@doug-s-nava I agree that a DOM query isn't ideal. An alternative to asserting that the mouseevent didn't click a dropdown wasn't immediately obvious to me. After checking in via slack, we've decided to create a backlog issue for this. Sharing here for visibility: https://github.com/HHS/simpler-grants-gov/issues/4797